### DISCUSSION TITLE
How do I shut down my dbus service properly?

### DISCUSSION THREAD

**Discussion by colazzo** [python-sdbus/python-sdbus#76] [General]
I generally start the app using asyncio.run(main())
How do we need to terminate the dbus service application correctly?

Is it necessary to call stop on DbusExportHandle explicitly?

**Comment by igo95862** [comment]
I believe you can call `close()` method of the bus to close connection. This should effectively stop the service.

**Reply by colazzo** [reply]
Do you mean just call it once on instance of SdBus, even if I have several exported objects (export_to_dbus)?

Can you give an example?

**Reply by igo95862** [reply]
If you used the `export_to_dbus` without specifying the bus then the default bus would be used. This means you can do `sdbus.get_default_bus().close()` which will close the default bus.

**Reply by colazzo** [reply]
@igo95862 thank you for quick responses!
